pkg/dogstatsd/intern.go

```diff
@@ -0,0 +1,25 @@
+package dogstatsd
+
+type stringInterner struct {
+	strings map[string]string
```

Comment:
Couldn't we use a `map[string]*string` here to always reuse the same pointer to string avoiding a bit of garbage?

Reply:
@remeh The string you get from the map will almost always be stored on the stack or in a struct that already has the space for the string (more precisely pointer to the "[]byte" + length).

@hush-hush While in theory doable I am not sure if a map `map[string]struct{}` would be practical. You want to get the version of the string that is in the map, not just confirm it's in the map or not. As far as I know there is no easy way to ask the map "give me your version of the key if my key is equal* to it".
*: here string are equal in term of their content, not in term of their address

Reply:
@arbll Yup you're right (I've even benched it)
And you're right, there is no way of getting the key value (maybe the range operator is returning it on a for but that's definitely not what we want to do).

Reply:
Indeed, that make sens. Thanks for the precision.